driver-async/src/main/com/mongodb/async/client/AggregateIterableImpl.java

```diff
@@ -185,53 +140,59 @@ public void batchCursor(final SingleResultCallback<AsyncBatchCursor<TResult>> ca
         return this;
     }
 
+    @Override
     @SuppressWarnings("deprecation")
-    private MongoIterable<TResult> execute() {
-        List<BsonDocument> aggregateList = createBsonDocumentList();
-        BsonValue outCollection = getAggregateOutCollection(aggregateList);
+    AsyncReadOperation<AsyncBatchCursor<TResult>> asAsyncReadOperation() {
+        List<BsonDocument> aggregateList = createBsonDocumentList(pipeline);
+        BsonValue outCollection = getOutCollection(aggregateList);
 
         if (outCollection != null) {
-            AggregateToCollectionOperation operation = new AggregateToCollectionOperation(namespace, aggregateList, writeConcern)
-                    .maxTime(maxTimeMS, MILLISECONDS)
-                    .allowDiskUse(allowDiskUse)
-                    .bypassDocumentValidation(bypassDocumentValidation)
-                    .collation(collation)
-                    .hint(hint == null ? null : hint.toBsonDocument(documentClass, codecRegistry))
-                    .comment(comment);
-            MongoIterable<TResult> delegated = new FindIterableImpl<TDocument, TResult>(new MongoNamespace(namespace.getDatabaseName(),
-                    outCollection.asString().getValue()), documentClass, resultClass, codecRegistry, primary(), readConcern,
-                    executor, new BsonDocument(), new FindOptions().collation(collation).maxAwaitTime(maxAwaitTimeMS, MILLISECONDS));
-            if (batchSize != null) {
-                delegated.batchSize(batchSize);
+            AggregateToCollectionOperation aggregateToCollectionOperation = createAggregateToCollectionOperation(aggregateList);
+            FindOperation<TResult> findOperation =
+                    new FindOperation<TResult>(new MongoNamespace(namespace.getDatabaseName(), outCollection.asString().getValue()),
+                            codecRegistry.get(resultClass))
+                            .readConcern(getReadConcern())
+                            .collation(collation)
+                            .maxAwaitTime(maxAwaitTimeMS, MILLISECONDS);
+            if (getBatchSize() != null) {
+                findOperation.batchSize(getBatchSize());
             }
-            return new AwaitingWriteOperationIterable<TResult, Void>(operation, executor, delegated);
```

Comment:
The AwaitingWriteOperationIterable class can be removed

Reply:
Done.